server/app/api/v1/endpoints/subscriptions.py

```
@@ -0,0 +1,2005 @@
+            billing_cycle_anchor = datetime.fromtimestamp(
+                stripe_subscription["billing_cycle_anchor"], tz=timezone.utc
+            )
+
+        subscription_metadata = subscription_data.subscription_metadata.copy() if subscription_data.subscription_metadata else {}
+        if subscription_data.billing_level == "organization":
+            if "rooftop_ids" not in subscription_metadata or not subscription_metadata.get("rooftop_ids"):
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="rooftop_ids must be provided in subscription_metadata for organization-level subscriptions. "
+                           "Please specify which rooftops should be included in this subscription.",
+                )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)